src/coreclr/src/pal/src/exception/signal.cpp

```diff
@@ -539,7 +539,7 @@ static void sigsegv_handler(int code, siginfo_t *siginfo, void *context)
             }
             else
             {
-                (void)write(STDERR_FILENO, StackOverflowMessage, sizeof(StackOverflowMessage) - 1);
+                (void)!write(STDERR_FILENO, StackOverflowMessage, sizeof(StackOverflowMessage) - 1);
```

Comment:
@janvorli, this one was from another gcc analysis, which we have disable globally (for both clang and gcc): https://github.com/dotnet/runtime/blob/8c0d7c1ebc5e9b2ffbb02170019f5b0c9bbf4262/eng/native/configurecompiler.cmake#L322-L325 I enable these locally for PAL only, and this fixed the debug build. However, there are more cases in PAL with Release build, which I have not touched as part of this PR.